mods/battle-royale/hook/lua/defaultunits.lua

```diff
@@ -0,0 +1,229 @@
+    end
+
+    if Multiplier == 1.5 then
+        T2StageTime = 420 -- 7 minute
+        T3StageTime = 840 -- 14 minute
+    end
+end
+
+InitStageTime()
+
+--- Depending on the time elapsed in the game, returns the id t1, t2 or t3 of the commander's beacon.
+function DefineBeaconId()
+    local result = 'uac1301'
+    local inGameTime = GetGameTimeSeconds()
+
+    if inGameTime > T2StageTime and inGameTime < T3StageTime then
+        result = 'xsc1501'
+    end
+
+    if inGameTime > T3StageTime then
+        result = 'xsc1301'
+    end
+
+    return result
+end
+
+function ResetUnitOrders(units)
+    if units and not table.empty(units) then
+        for _, unit in units do
+            if not unit.Dead then
+                IssueStop({unit})
+                IssueClearCommands({unit})
+            end
+        end
+    end
+end
+
+--- Creates a beacon at the location of the death of the ACU.
+--- The beacon contains a list of the killed player's units.
+--- The player who captures the beacon will receive all the units of the killed player.
+--- @param units - list of killed player's units
+--- @param xPos, yPos, zPos - coordinates of the destroyed ACU
+function CreateCommanderBeacon(units, xPos, yPos, zPos)
+    local beaconId = DefineBeaconId()
+    local beacon = CreateUnitHPR(beaconId, "NEUTRAL_CIVILIAN", xPos, yPos, zPos, 0, 0, 0)
+
+    -- makes beacon invulnerable for 4 seconds.
+    beacon:SetCanTakeDamage(false)
+    ForkThread(function(unit)
+        if not unit then
+            return
+        end
+        WaitSeconds(4)
+        unit:SetCanTakeDamage(true) end,beacon)
+
+    -- spawn effect
+    local effects = EffectUtils.CreateEffects(beacon, 1, CarePackageTeleportIn)
+
+    -- when reclaimed, everything dies
+    local function OnReclaimed(self)
+        for k, unit in units do
+            if not unit.Dead then
+                unit:Kill()
+            end
+        end
+
+        beacon = nil
+    end
+
+    -- when killed, everything dies
+    local function OnKilled(self)
+        for k, unit in units do
+            if not unit.Dead then
+                unit:Kill()
+            end
+        end
+
+        beacon = nil
+    end
+
+
+    -- when captured, everything is given
+    local function OnCaptured(old, new)
+
+        local effects = EffectUtils.CreateEffects(beacon, 1, CarePackageDestroyed)
+        beacon:Destroy()
+        TransferUnitsOwnership(units, new.Army)
+
+        beacon = nil
+    end
+
+    beacon:AddUnitCallback(OnReclaimed, "OnReclaimed")
+    beacon:AddUnitCallback(OnKilled, "OnKilled")
+    beacon:AddOnCapturedCallback(OnCaptured)
+end
+
+--- Iterates through the list of units, and destroys unfinished units.
+--- @param units - list of player units
+function DestroyUnfinishedUnits(units)
+    for _, unit in units do
+        if not unit.Dead and unit:GetFractionComplete() ~= 1 then
+            unit:Destroy()
+        end
+    end
+end
+
+function TransferUnitsToNeutralArmyAndReturnNewList(units)
```

Comment:
There is no description because the name of the function fully reflects the functionality and the function does not contain a non-obvious implementation.